ddl/reorg.go

```diff
@@ -742,6 +745,35 @@ func getReorgInfoFromPartitions(ctx *JobContext, d *ddlCtx, rh *reorgHandler, jo
 	return &info, nil
 }
 
+func (r *reorgInfo) deleteReorgMeta(pool *sessionPool) error {
+	if len(r.elements) == 0 {
+		return nil
+	}
+	se, err := pool.get()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer pool.put(se)
+
+	sess := newSession(se)
+	err = sess.begin()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	txn, err := sess.txn()
+	if err != nil {
+		sess.rollback()
+		return errors.Trace(err)
+	}
+	rh := newReorgHandler(meta.NewMeta(txn), sess, variable.EnableConcurrentDDL.Load())
+	err = rh.RemoveDDLReorgHandle(r.Job, r.elements)
```

Comment:
If this `err` is not nil, do we need to do `rollback`?

Reply:
I changed the implementation, so it will remove entries for the job in a separate transaction when the job is finished. In case the cleanup fails, it will now only log it.